symengine/fields.cpp

```diff
@@ -0,0 +1,267 @@
+#include <symengine/fields.h>
+#include <symengine/add.h>
+#include <symengine/constants.h>
+#include <symengine/mul.h>
+#include <symengine/pow.h>
+
+namespace SymEngine
+{
+GaloisField::GaloisField(const RCP<const Symbol> &var, GaloisFieldDict &&dict)
+    : UPolyBase(var, std::move(dict))
+{
+    SYMENGINE_ASSERT(is_canonical(poly_))
+}
+
+bool GaloisField::is_canonical(const GaloisFieldDict &dict) const
+{
+    // Check if dictionary contains terms with coeffienct 0
+    if (dict.modulo_ <= integer_class(0))
+        return false;
+    for (auto iter : dict.dict_)
+        if (iter.second == 0)
+            return false;
+    return true;
+}
+
+std::size_t GaloisField::__hash__() const
+{
+    std::hash<std::string> hash_string;
+    std::size_t seed = GALOISFIELD;
+
+    seed += hash_string(this->var_->get_name());
+    for (const auto &it : poly_.dict_) {
+        std::size_t temp = GALOISFIELD;
+        hash_combine<unsigned int>(temp, it.first);
+        hash_combine<long long int>(temp, mp_get_si(it.second));
+        seed += temp;
+    }
+    return seed;
+}
+
+int GaloisField::compare(const Basic &o) const
+{
+    const GaloisField &s = static_cast<const GaloisField &>(o);
+
+    if (poly_.size() != s.poly_.size())
+        return (poly_.size() < s.poly_.size()) ? -1 : 1;
+
+    int cmp = unified_compare(var_, s.var_);
+    if (cmp != 0)
+        return cmp;
+
+    cmp = unified_compare(poly_.modulo_, s.poly_.modulo_);
+    if (cmp != 0)
+        return cmp;
+
+    return unified_compare(poly_.dict_, s.poly_.dict_);
```

Comment:
shouldn't it use
`return unified_compare(std::move(poly_.dict_), std::move(s.poly_.dict_));`
?